RemoteRenderingBackend timeouts are inconsistent

kkinnunen-apple opened the pull request:
#### 6e42b6fe8fff8ef8ede6b382e291b0b97f8e42cd
<pre>
RemoteRenderingBackend timeouts are inconsistent
<a href="https://bugs.webkit.org/show_bug.cgi?id=258719">https://bugs.webkit.org/show_bug.cgi?id=258719</a>
rdar://111559663

Reviewed by Simon Fraser.

Use the same timeout for both sends from RemoteRenderingBackend
as well as sends from RemoteDisplayListRecorderProxy.

Use the same timeout for waits as for sends.

* Source/WebKit/WebProcess/GPU/graphics/RemoteDisplayListRecorderProxy.cpp:
(WebKit::RemoteDisplayListRecorderProxy::send):
(WebKit::RemoteDisplayListRecorderProxy::sendSync):
* Source/WebKit/WebProcess/GPU/graphics/RemoteDisplayListRecorderProxy.h:
* Source/WebKit/WebProcess/GPU/graphics/RemoteRenderingBackendProxy.cpp:
(WebKit::RemoteRenderingBackendProxy::send):
(WebKit::RemoteRenderingBackendProxy::sendSync):
* Source/WebKit/WebProcess/GPU/graphics/RemoteRenderingBackendProxy.h:
(WebKit::RemoteRenderingBackendProxy::send): Deleted.
(WebKit::RemoteRenderingBackendProxy::sendSync): Deleted.

Canonical link: <a href="https://commits.webkit.org/265720@main">https://commits.webkit.org/265720@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ace08853a73e97c8a8064698a5993e2fe896a218

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/11712 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/11911 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/12279 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/13356 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/11156 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/14300 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/11892 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/14022 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/11876 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/12717 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/9938 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/13777 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/10009 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/10634 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/17764 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/11086 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/10788 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/13959 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/11189 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/9236 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/10366 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/10513 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2817 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/14646 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/11046 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebKit/WebProcess/GPU/graphics/RemoteDisplayListRecorderProxy.cpp:
```diff
@@ -74,7 +74,9 @@ ALWAYS_INLINE void RemoteDisplayListRecorderProxy::send(T&& message)
-    m_renderingBackend->streamConnection().send(WTFMove(message), m_destinationBufferIdentifier, defaultSendTimeout);
+    auto result = m_renderingBackend->streamConnection().send(WTFMove(message), m_destinationBufferIdentifier, RemoteRenderingBackendProxy::defaultTimeout);
+    if (UNLIKELY(result != IPC::Error::NoError))
+        RELEASE_LOG(RemoteLayerBuffers, "%p RemoteDisplayListRecorderProxy::sync - failed, name:%" PUBLIC_LOG_STRING ", error:%" PUBLIC_LOG_STRING, this, IPC::description(T::name()), IPC::errorAsString(result));
```
This should say `RemoteDisplayListRecorderProxy::send()` I think?

smfr commented on Source/WebKit/WebProcess/GPU/graphics/RemoteRenderingBackendProxy.cpp:
```diff
@@ -98,6 +98,23 @@ void RemoteRenderingBackendProxy::ensureGPUProcessConnection()
+{
+    auto result = streamConnection().send(WTFMove(message), renderingBackendIdentifier(), defaultTimeout);
+    if (UNLIKELY(result != IPC::Error::NoError))
+        RELEASE_LOG(RemoteLayerBuffers, "%p RemoteRenderingBackendProxy::send - failed, name:%" PUBLIC_LOG_STRING ", error:%" PUBLIC_LOG_STRING, this, IPC::description(T::name()), IPC::errorAsString(result));
```
Can we log `"[pageProxyID=%" PRIu64 ", webPageID=%" PRIu64 ", renderingBackend=%" PRIu64 "]` like the other calls, so it's easy to correlate with other objects in logging? Might want a shared function for the RELEASE_LOG to make this easier (or macros).

kkinnunen-apple committed:
RemoteRenderingBackend timeouts are inconsistent
https://bugs.webkit.org/show_bug.cgi?id=258719
rdar://111559663

Reviewed by Simon Fraser.

Use the same timeout for both sends from RemoteRenderingBackend
as well as sends from RemoteDisplayListRecorderProxy.

Use the same timeout for waits as for sends.

* Source/WebKit/WebProcess/GPU/graphics/RemoteDisplayListRecorderProxy.cpp:
(WebKit::RemoteDisplayListRecorderProxy::send):
(WebKit::RemoteDisplayListRecorderProxy::sendSync):
* Source/WebKit/WebProcess/GPU/graphics/RemoteDisplayListRecorderProxy.h:
* Source/WebKit/WebProcess/GPU/graphics/RemoteRenderingBackendProxy.cpp:
(WebKit::RemoteRenderingBackendProxy::send):
(WebKit::RemoteRenderingBackendProxy::sendSync):
* Source/WebKit/WebProcess/GPU/graphics/RemoteRenderingBackendProxy.h:
(WebKit::RemoteRenderingBackendProxy::send): Deleted.
(WebKit::RemoteRenderingBackendProxy::sendSync): Deleted.

Canonical link: https://commits.webkit.org/265720@main
